src/main/java/soolutions/hometaskapp/common/Amount.java

```diff
@@ -8,26 +8,33 @@
   private final String currency;
 
   public Amount(double value, String currency) {
-    if (value < 0) {
-      throw new InvalidAmountException(value);
-    }
-
     this.value = value;
     this.currency = currency;
   }
 
+  public Amount zero() {
```

Comment:
I didn't found a good way to implement this as static method in current iteration because `currency` is encapsulated, so the only way to do this with static is:
```c#
public static Amount zero( Amount amount ) { return new Amount( 0, amount.currency ); }
```

And the calls:
```c#
var zero1 = amount.zero();
var zero2 = Amount.zero( amount );
```

Do you think the second option is preferable?

Reply:
Maybe it'll be better to name method like:
`Amount.zeroOf( amount )`?

Reply:
`Amount.zeroOf( "USD" )` makes more sense.

Reply:
> Amount.zeroOf( "USD" ) makes more sense.

For example, I want to check whether specified amount is less than zero:
```c#
void check( Amount amount ) {
    var zero = Amount.zeroOf( ??? );
}
```

Reply:
Like I said guys, the amount object is fucked. We need to replace with Currency for this exact sort of issue.